OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -122,19 +125,34 @@ public SaveMapLogic(Widget widget, Action onExit, Map map, EditorActorLayer edit
 				map.Description = description.Text;
 				map.Author = author.Text;
 				map.Visibility = (MapVisibility)Enum.Parse(typeof(MapVisibility), visibilityDropdown.Text);
-				map.ActorDefinitions = editorActorLayer.Save();
-				map.PlayerDefinitions = editorActorLayer.Players.ToMiniYaml();
+
+				if (actorDefinitions != null)
+					map.ActorDefinitions = actorDefinitions;
+
+				if (playerDefinitions != null)
+					map.PlayerDefinitions = playerDefinitions;
+
 				map.RequiresMod = Game.ModData.Manifest.Mod.Id;
 
 				var combinedPath = Platform.ResolvePath(Path.Combine(directoryDropdown.Text, filename.Text + fileTypes[typeDropdown.Text]));
+
+				// Invalidate the old map metadata
+				if (map.Uid != null)
```

Comment:
You should only do that if the old path is != the new path. In a situation where I want to modify an existing map and save it under a different name, the old map would otherwise disappear.